workshop/exercise-7/README.md

```diff
@@ -0,0 +1,193 @@
+## (OPTIONAL, FOR ADVANCED USERS) Build and Deploy our Knative Application
+
+We've seen how to deploy an application from a container that already exists, so let's build the container image ourselves from source code. We'll be using Tekton Pipelines, a project providing Kubernetes-style resources for declaring CI/CD pipelines. We'll fetch our source code from a github repository, build a docker image from it, push it to a private container registry on IBM Cloud, and then ultimately get a URL to access our application.
+
+### Tekton Components Overview
+
+First, let's understand a few of the Tekton components we will be using in this exercise. A `Task` defines work that needs to be executed, and is made up of a series of `steps` that will be executed sequentially.
+
+A `TaskRun` will run the `Task` that you defined. You can then view output from the `TaskRun`.
```

Comment:
Up to you, but if you want to keep it simple you could inline the steps into the TaskRun and create one object instead of two.

Reply:
Will keep two objects for now to mirror what's going on in the tekton getting started project & enable reuse of components.